### PR TITLE
Fix for account creation

### LINKED ISSUE #232: Account creation error in client
**Describe the bug**
I the client, creating an account returns an error (but still creates the account)

**To Reproduce**
1. Go to client homepage
2. Click on Account icon
2. Click on "Create An Account"
3. Fill the form and submit it
4. See error "GraphQL error: Cannot destructure property `input` of 'undefined' or 'null'"

**Expected behavior**
Have a creation success message (or a failure message) in the user interface

**Desktop:**
 - OS: Ubuntu
 - Browser Chrome
 - Version Google Chrome 71.0.3578.80

### PR DESCRIPTION
**Pull request by mateuszwozniak**
This pr fixes #232 - sign up user correctly after account creation.

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
It turned out that after refactoring we've been calling `signIn()` from `signUp` with incorrect parameters. That's fixed now.

### CODE REVIEW

File: packages/falcon-magento2-api/src/index.test.js [hunk 1]
```diff
@@ -131,4 +132,49 @@ describe('Magento2Api', () => {
     // null should not not be converted
     expect(api.convertPathToUrl(null)).toBe(null);
   });
+
+  it('Should not call signIn() after account creation if autoSignIn flag was set to false', async () => {
+    // mock signIn resolver - just return promise that resolves to true
+    api.signIn = sinon.fake.resolves(true);
```
Comment (Borales) [inline packages/falcon-magento2-api/src/index.test.js]: @mateuszwozniak please use https://jestjs.io/docs/en/mock-functions#using-a-mock-function

```suggestion
    api.signIn = jest.fn(() => Promise.resolve(true));
```

Comment (mateuszwozniak) [reply packages/falcon-magento2-api/src/index.test.js]: done

File: packages/falcon-magento2-api/src/index.test.js [hunk 2]
```diff
@@ -131,4 +131,49 @@ describe('Magento2Api', () => {
...
+      .reply(200, {});
+
+    const resp = await api.signUp(
+      {},
+      {
+        input: { firstname: 'Test', lastname: 'Test', email: 'test@test.com', password: 'Deity123', autoSignIn: false }
+      }
+    );
+
+    expect(api.signIn.mock.calls.length).toBe(0);
```
Comment (Borales) [inline packages/falcon-magento2-api/src/index.test.js]:
```suggestion
    expect(api.signIn).toHaveBeenCalledTimes(0);
```

File: packages/falcon-magento2-api/src/index.test.js [hunk 3]
```diff
@@ -131,4 +131,49 @@ describe('Magento2Api', () => {
...
+    nock(URL)
+      .post(createMagentoUrl('/guest-carts'))
+      .reply(200, {});
+
+    const resp = await api.signUp(
+      {},
+      { input: { firstname: 'Test', lastname: 'Test', email: 'test@test.com', password: 'Deity123', autoSignIn: true } }
+    );
+
+    expect(api.signIn.mock.calls.length).toBe(1);
```
Comment (Borales) [inline packages/falcon-magento2-api/src/index.test.js]:
```suggestion
    expect(api.signIn.mock).toHaveBeenCalledTimes(1);
```

File: packages/falcon-magento2-api/src/index.test.js [hunk 4]
```diff
@@ -131,4 +131,49 @@ describe('Magento2Api', () => {
...
+    expect(api.signIn.mock.calls[0][1]).toEqual({ input: { email: 'test@test.com', password: 'Deity123' } });
```
Comment (Borales) [inline packages/falcon-magento2-api/src/index.test.js]:
```suggestion
    expect(api.signIn).toHaveBeenCalledWith({}, { input: { email: 'test@test.com', password: 'Deity123' } });
```

File: packages/falcon-magento2-api/src/index.test.js (hunk 2 above)
Comment (mateuszwozniak) [reply packages/falcon-magento2-api/src/index.test.js]: I seriously prefer sinon :P

**Pull request merged**